scripts/externalTests/test_helpers.py

```diff
@@ -0,0 +1,284 @@
+# pragma pylint: disable=import-error,wrong-import-position
+PROJECT_ROOT = Path(__file__).parents[2]
+sys.path.insert(0, f"{PROJECT_ROOT}/scripts/common")
+
+from git_helpers import git_commit_hash
+
+SOLC_FULL_VERSION_REGEX = re.compile(r"^[a-zA-Z: ]*(.*)$")
+SOLC_SHORT_VERSION_REGEX = re.compile(r"^([0-9.]+).*\+|\-$")
+
+evm_version = os.environ.get("DEFAULT_EVM")
+CURRENT_EVM_VERSION: str = evm_version if evm_version is not None else "shanghai"
+
+AVAILABLE_PRESETS: List[str] = [
+    "legacy-no-optimize",
+    "ir-no-optimize",
+    "legacy-optimize-evm-only",
+    "ir-optimize-evm-only",
+    "legacy-optimize-evm+yul",
+    "ir-optimize-evm+yul",
+]
+
+
+@dataclass
+class TestConfig:
+    name: str
+    repo_url: str
+    ref_type: str
+    ref: str
+    build_dependency: str = field(default="nodejs")
+    compile_only_presets: List[str] = field(default_factory=list)
+    settings_presets: List[str] = field(default_factory=lambda: AVAILABLE_PRESETS)
+    evm_version: str = field(default=CURRENT_EVM_VERSION)
+
+    def selected_presets(self):
+        return set(self.compile_only_presets + self.settings_presets)
+
+
+class TestRunner(metaclass=ABCMeta):
+    config: TestConfig
+    solc_binary_type: str
+    solc_binary_path: Path
+
+    def __init__(self, argv, config: TestConfig):
+        args = parse_command_line(f"{config.name} external tests", argv)
+        self.config = config
+        self.solc_binary_type = args.solc_binary_type
+        self.solc_binary_path = args.solc_binary_path
+        self.env = os.environ.copy()
+        self.tmp_dir = mkdtemp(prefix=f"ext-test-{config.name}-")
+        self.test_dir = Path(self.tmp_dir) / "ext"
+
+    def setup_solc(self) -> str:
+        if self.solc_binary_type == "solcjs":
+            # TODO: add support to solc-js # pylint: disable=fixme
+            raise NotImplementedError()
+        print("Setting up solc...")
+        solc_version_output = subprocess.getoutput(f"{self.solc_binary_path} --version").split(":")[1]
+        return parse_solc_version(solc_version_output)
+
+    @staticmethod
+    def on_local_test_dir(fn):
+        """Run a function inside the test directory"""
+
+        def f(self, *args, **kwargs):
+            assert self.test_dir is not None
+            os.chdir(self.test_dir)
+            return fn(self, *args, **kwargs)
+
+        return f
+
+    def setup_environment(self):
+        """Configure the project build environment"""
+        print("Configuring Runner building environment...")
+        replace_version_pragmas(self.test_dir)
+
+    @on_local_test_dir
+    def clean(self):
+        """Clean temporary directories"""
+        rmtree(self.tmp_dir)
+
+    @on_local_test_dir
+    def compiler_settings(self, _: List[str]):
+        # TODO: default to hardhat # pylint: disable=fixme
+        raise NotImplementedError()
+
+    @on_local_test_dir
+    def compile(self, _: str):
+        # TODO: default to hardhat # pylint: disable=fixme
```

Comment:
I would not really pick a default here. Hardhat-specific code should go into Hardhat runner.